### DISCUSSION TITLE
Generate ElasticSearch Query

### DISCUSSION THREAD

**Discussion by vvillait88** [react-querybuilder/react-querybuilder#525] [Ideas]
Allow for the generation of ElasticSearch Query.

**Comment by jakeboone02** [comment]
Thanks @vvillait88. I've looked into this before and had a hard time mapping the default operators and the query structure to ElasticSearch syntax (which I've never used before).

Could you provide some examples of RQB queries and the corresponding ElasticSearch format?

**Reply by vvillait88** [reply]
@jakeboone02 React Awesome Query Builder is doing this, so maybe it would be easier to dig into how they are and replicating: https://github.com/ukrbublik/react-awesome-query-builder

**Reply by jakeboone02** [reply]
Yeah, good point. Their query structure is a little different but I'll take a look.

**Reply by jakeboone02** [reply]
Hey @vvillait88, I know it's been a while but I finally got around to implementing an ElasticSearch export (#583). I don't know how to test it, I was mostly just going by `react-awesome-query-builder`'s implementation.

If you still need this, or if you could spare some time to look at it, you can test it out in [this sandbox](https://codesandbox.io/p/sandbox/react-querybuilder-ci-forked-2n54n4). The ElasticSearch output shows up just below the query options.

I'm particularly interested in the validity of the output for rules with `valueSource: "field"` (since they generate a `script` string), especially when the `operator` is "contains"/"beginsWith"/"endsWith" or one of the negated counterparts. I _think_ it's done right based on the ElasticSearch docs, but I can't be 100% sure until it's tested in a real environment.
